carservice/views.py

```diff
@@ -1,55 +1,130 @@
-from django.shortcuts import render, redirect
-from django.views import generic, View
+from django.shortcuts import render
+from django.views import View
 from django.urls import reverse_lazy
 from django.views.generic.edit import CreateView
-
 from .models import Car, Offer, Rent
-from django.utils import timezone
+from django.contrib.auth.mixins import LoginRequiredMixin, UserPassesTestMixin
+from django.utils.timezone import now
 
 
-class CarCreateView(CreateView):
+class CarCreateView(LoginRequiredMixin, UserPassesTestMixin, CreateView):
+    login_url = 'home'
+    redirect_field_name = 'redirect_to'
     model = Car
-    fields = ['serial_number', 'car_mileage', 'car_model', 'car_year']
+    fields = ['serial_number', 'car_mileage', 'car_brand', 'car_model', 'year_of_prod']
     template_name = 'car_create.html'
     success_url = reverse_lazy('car_read')
 
+    def get_initial(self):
+        initial = super().get_initial()
+        initial['year_of_prod'] = 'YYYY-MM-DD'
+        return initial
+
+    def test_func(self):
+        return self.request.user.is_authenticated
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        return super().form_valid(form)
+
+
+class CarReadView(LoginRequiredMixin, UserPassesTestMixin, View):
+    def test_func(self):
+        return self.request.user.is_authenticated
+
+    def form_valid(self, form):
+        form.instance.user = self.request.user
+        return super().form_valid(form)
+
+    def get_queryset(self):
+        if self.request.user.is_superuser:
+            return Car.objects.all()
+        else:
+            return Car.objects.filter(user=self.request.user)
 
-class CarReadView(View):
     def get(self, request):
+        cars = self.get_queryset()
         return render(
             request, template_name='car_read.html',
-            context={'cars': Car.objects.all()}
+            context={'cars': cars}
         )
 
 
-class OfferCreateView(CreateView):
+class OfferCreateView(LoginRequiredMixin, UserPassesTestMixin, CreateView):
     model = Offer
     fields = ['car', 'description', 'price']
     template_name = 'offer_create.html'
     success_url = reverse_lazy('offer_read')
 
+    def test_func(self):
```

Comment:
lepiej nie nazywać 'test_' albo '...._test'  bo jak napiszesz testy dla tego projektu prawdopodobnie ci to zweryfikuje jako test do sprawdzenia, lepiej nazwać "check_auth" naprzykład.

Reply:
done